File: evaluation/benchmark/bintree.py
#!/usr/bin/python

import sys

def bintree(depth):
  if depth == 0:
    return None
  else:
    return (bintree(depth-1),bintree(depth-1))

def countnodes(tree):
  if tree is not None:
    return 1 + countnodes(tree[0]) + countnodes(tree[1])
  else:
    return 0

n = 16
if (len(sys.argv) >= 2):
  n = int(sys.argv[1])

tree = bintree(n)
count = countnodes(tree)
print count

File: evaluation/benchmark/bintree.pl
#!/usr/bin/perl -w

sub bintree
{
  my $depth = $_[0];

  if ($depth == 0) {
    return undef;
  }
  else {
    return [ bintree($depth-1), bintree($depth-1) ];
  }
}

sub countnodes
{
  my ($tree) = @_;
  if ($tree) {
    return 1+countnodes($tree->[0])+countnodes($tree->[1]);
  }
  else {
    return 0;
  }
}

my $n = ($#ARGV < 0) ? 16 : $ARGV[0];
my $tree = bintree($n);
my $count = countnodes($tree);
print "$count\n";
